Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1429,6 +1468,114 @@ TEST_F(SerializerTest, EncodesResumeTokens) {
   ExpectRoundTrip(model, proto);
 }
 
+TEST_F(SerializerTest, EncodesListenRequestLabels) {
+  core::Query q = Query("docs");
+
+  std::map<QueryPurpose, std::unordered_map<std::string, std::string>>
+      purpose_to_label = {
+          {QueryPurpose::Listen, {}},
+          {QueryPurpose::LimboResolution,
+           {{"goog-listen-tags", "limbo-document"}}},
+          {QueryPurpose::ExistenceFilterMismatch,
+           {{"goog-listen-tags", "existence-filter-mismatch"}}},
+      };
+
+  for (const auto& p : purpose_to_label) {
+    QueryData model(q, 1, 0, p.first);
+
+    auto result = serializer.EncodeListenRequestLabels(model);
+    EXPECT_EQ(result, p.second);
+  }
+}
+
+TEST_F(SerializerTest, DecodesListenResponseWithTargetChange) {
+  WatchTargetChange model(WatchTargetChangeState::Added, {1, 2},
```

Comment:
There are [3 tests](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Example/Tests/Remote/FSTSerializerBetaTests.mm#L870-L908) involving `WatchTargetChange` in Objective-C, can you please port them all?